internal/toproto6/diagnostics_test.go

```diff
@@ -10,6 +10,42 @@ import (
 	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
+func TestDiagnosticSeverity(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		severity diag.Severity
+		expected tfprotov6.DiagnosticSeverity
+	}{
+		"error": {
+			severity: diag.SeverityError,
+			expected: tfprotov6.DiagnosticSeverityError,
+		},
+		"invalid": {
+			severity: diag.SeverityInvalid,
+			expected: tfprotov6.DiagnosticSeverityInvalid,
+		},
+		"warning": {
+			severity: diag.SeverityWarning,
+			expected: tfprotov6.DiagnosticSeverityWarning,
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
```

Comment:
Just wondering why the `name` and `testCase` vars that are declared in the `for` statement are not used directly?

Reply:
I was going to ask the same question.
Then I tried: https://go.dev/play/p/JFh-5r_DgnS

I ... don't like this thing of Go.
The closure should capture the value, not forcing this so that the value gets copied in a separate variable each time.

Well, I guess I'd expect Go to do it by itself.